src/tainting/Taint_lval_env.ml

```diff
@@ -480,3 +481,9 @@ let to_string taint_to_str
       pending_propagation_dests "[PENDING PROPAGATION DESTS]"
 
 let seq_of_tainted env = LvalMap.to_seq env.tainted
+
+let find_tainted_lvals_of_common_base { tainted; _ } { IL.base; _ } =
+  LvalMap.fold
+    (fun ({ base = base'; _ } as lval) _ l ->
+      if base =*= base' then lval :: l else l)
```

Comment:
Better define `compare_base` based on `IL.compare_name`. Polymorphic compare is dangerous and it has bitten us before!

Reply:
done.